services/datamanager/builtin/capture/capture.go

```diff
@@ -124,6 +138,75 @@ func (c *Capture) newCollectors(collectorConfigsByResource CollectorConfigsByRes
 	return newCollectors
 }
 
+func (c *Capture) mongoSetup(ctx context.Context, mc MongoConfig) *mongo.Collection {
+	if err := mc.validate(); err != nil {
+		c.logger.Warn("mongo_config is invalid: %s", err.Error())
+		return nil
+	}
+	// closes collectors and old connection if there is one and nils out mongo connection struct members
+	c.closeNoMongoMutex(ctx)
+	// Use the SetServerAPIOptions() method to set the Stable API version to 1
+	serverAPI := options.ServerAPI(options.ServerAPIVersion1)
+	// Create a new client and connect to the server
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(mc.URI).SetServerAPIOptions(serverAPI))
+	if err != nil {
+		c.logger.Warn("failed to create mongo connection with mongo_config.url")
+		return nil
+	}
+	c.mongoClient = client
+	database := defaultIfZeroVal(mc.Database, defaultMongoDatabaseName)
+	collection := defaultIfZeroVal(mc.Collection, defaultMongoCollectionName)
+	c.mongoCollection = c.mongoClient.Database(database).Collection(collection)
+	c.mongoConfig = &mc
+	return c.mongoCollection
+}
+
+// mongoReconfigure
+// it shuts down the collectors when on the mongo client is no longer being valid based on the new config
+// and attempts to create a new mongo client when the new config perscribes one.
+// returns a *mongo.Collection when the client is valid and nil when it is not.
+func (c *Capture) mongoReconfigure(ctx context.Context, mc *MongoConfig) *mongo.Collection {
+	c.mongoMU.Lock()
+	defer c.mongoMU.Unlock()
+
+	switch {
+	case c.mongoClient == nil && c.mongoConfig == nil && mc == nil:
```

Comment:
I found this function really hard to follow - for readability, can we redeclare c.mongoConfig and mc as old and new config, and then simplify these cases into just a couple if statements? For example:

```go
oldConfig := c.mongoConfig
newConfig := mc

if newConfig != nil {
    if oldConfig.Equals(newConfig) {
        return c.mongoCollection
    }
    return c.mongoSetup(ctx, *newConfig)
}

c.mongoConfig = nil // reset cached config to nil
if c.mongoClient != nil {
     c.closeNoMongoMutex(ctx)
}
return nil
```

Not 100% sure if this logic is equivalent so please check me on that!

Reply:
I'll see if I can compress this.

Reply:
I've compressed it. LMK what you think